covid_graphs/covid_graphs/simulation_report.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List
+
+from .pb.simulation_results_pb2 import SimulationResults
+
+
+@dataclass
+class SimulationReport():
+    daily_positive: List[List[int]]
+    daily_infected: List[List[int]]
+    deltas: List[float]
+    b0: int
+    prefix_length: int
+    error: float
+    # We want to store either alpha or gamma2. We could have two classes, one having alpha and the
+    # other gamma2 (in C++ I would make a template). Don't know what's best practice in these
+    # situations in Python.
```

Comment:
We have `oneof` in the proto and I'd like to simulate the same behavior: https://github.com/lukipuki/COVID-19-simulation/blob/master/src/simulation_results.proto#L6-L9

Reply:
I think "What does `oneof` in the proto try to achieve?" is the right question to ask.

If I understand correctly, it seems there are two types of simulation types. Each type has a different set of parameters, in the future maybe there will be one with two float parameters, who knows.

If that's the case, you could define the two simulation types separately:
```
@dataclass
class PolynomialGrowth:
    alpha: float

@dataclass
class ExponentialGrowth:
    gamma2: float
```
and then have
```
class SimulationReport:
    ...
    simulation_params: Union[PolynomialGrowth, ExponentialGrowth]
```

Bottomline: I think that would be an overkill, I'd keep what you have if we are just printing the params. Or B) above if we are using them in python code :)

Reply:
Since I've added growth_type, we don't *really* need inheritance or anything, so I'll keep it as it if for now. At some point I might add inheritance or optional, but it's very low priority right now.